### ISSUE TITLE
Test enhancements for Integration Testing

### ISSUE THREAD

**Issue by jzbmw** [eclipse-tractusx/item-relationship-service#226]
**As** Open Source Developer ,
**I want** to be able to test my developments on the localdeployment ,
**so that** the behavior is stable within the project.

## Link 
- is blocked by #375 
- child of https://github.com/eclipse-tractusx/sig-release/issues/650

## Hints / Details

- Integration Testing is conducted on helm deloyment ( local, github-runner) 
  - PCWM Way: Github Runners ( slow solution )
-  Use Gihub local runner with HELM tests 
- sync the feature files for executing the tests with cli
- Currently the feature files are managed by Xray
- After e2e testing for R 24.05 transfer and maintain feature files in the codebase

## Acceptance Criteria
- [x] Setup the testing environment on umbrella helm deployment
- [x] Execute cucumber from github workflows and via cli 
- [x] Execute tavern from github workflows and via cli
- [x] Clear description in documentation what services are mocked and which not
- [x] ARC42 documentation Test Management in Cross-Cutting concepts   

## Testing 
- Check documentation 
- Start irs local deployment with smoke tests 

###  Mocked 
- DIM 
- BPN to EDC Service 

### Not Mocked 
- iDP 
- dDTR 
- EDC 
- EDC  Discovery Service 
- Discovery Finder 
 
### Not Clear 
- Semantic Hub 
## Out of Scope

- building the umbrella helm chart #312

**Comment by ds-jhartmann** [comment]
Created the testing environment in this PR https://github.com/eclipse-tractusx/item-relationship-service/pull/732

**Comment by ds-jhartmann** [comment]
Fixed failing tests in https://github.com/eclipse-tractusx/item-relationship-service/pull/856. 
Ready for PO acceptance
